engine/table/src/main/java/io/deephaven/engine/table/impl/util/RuntimeMemory.java

```diff
@@ -172,10 +181,7 @@ public void read(final Sample buf) {
                 }
             }
         }
-        buf.freeMemory = snapshot.lastFreeMemory;
-        buf.totalMemory = snapshot.lastTotalMemory;
-        buf.totalCollections = snapshot.totalCollections;
-        buf.totalCollectionTimeMs = snapshot.totalCollectionTimeMs;
+        snapshot.readInto(buf);
         if (logInterval > 0 && now >= snapshot.nextLog) {
             synchronized (this) {
                 if (now >= currSnapshot.nextLog) {
```

Comment:
This part is kind of weird, I think to clean it up:
- The JSDoc for `logInterval` should specify that a value of `0` means "Don't log at all" (as opposed to logging every time)
- Checking against snapshot.nextCheck twice... I think it would be clearer inside the synchronized blocks if the if check was just `if (snapshot == currSnapshot)`... right now it's possible for one thread to replace the snapshot, but the logging to be logged by the next thread that comes in (which could replace the snapshot again even). Not that it _really_ matters.

Reply:
The problem is you might not replace the snapshot, because another thread won.  An else of the now >= currSnapshot.nextCheck likely should set snapshot to currSnapshot.

Reply:
Happy to improve, but in future PRs?